### DISCUSSION TITLE
Extensibility of Rodauth : where to put new fields like "role" ?

### DISCUSSION THREAD

**Discussion by bdavidxyz** [janko/rodauth-rails#89] [General]
Good Morning. If I have to add a column like "role" (for later use with Pundit), would it be better to put on an external table like "Profile", or am I allowed to put it inside the Rodauth "Account" table, without the risk of breaking anything ? Best regards.

**Comment by janko** [comment]
Adding a new column to the `accounts` table shouldn't break anything, but I still recommend using a separate table such as `roles`. For one, this maintains a separation of concerns between authentication and authorization. But more importantly, it allows accounts to have multiple roles, which can be useful later on (I've heard people recommend modeling roles in this way).

**Reply by twe4ked** [reply]
If putting something like this in a separate table, is it possible to `JOIN` to that table in the `current_account` helper method to prevent the need for extra queries? I'm a bit unclear on where the best place for that kind of override is.

My use case is slightly different to “role”, I'm looking at adding an `archived` flag that we want to check on every request–it's also slightly different to the generic `close_account` feature.

**Reply by janko** [reply]
I believe you could override the `account_ds` method to do this:

```rb
class RodauthMain < Rodauth::Rails::Auth
  configure do
    # ...
    auth_class_eval do
      def account_ds(*)
        super.join(:your_table, account_id: :id)
      end
    end
  end
end
```

Hopefully Active Record would store the extra attributes, and you would be able to call `current_account[:role]`.

**Reply by kyle-rader** [reply]
@janko I'm also looking to extend the `Account` model with a few more application specific fields that I want to set as a part of account creation so that DB constraints are all checked within the same transaction, and only 1 query is needed. Is there a way to customize the set of expected params for creating an account?

**Reply by twe4ked** [reply]
Janko, sorry I forgot to reply earlier, thanks for your response!

**Reply by kyle-rader** [reply]
Ah I found the documentation I was looking for at http://rodauth.jeremyevans.net/rdoc/files/doc/guides/registration_field_rdoc.html. This isn't actually something `rodauth-rails` cares about, it's configuration at the `rodauth` level :)
